nsxt/resource_nsxt_policy_ipsec_vpn_service.go

```diff
@@ -0,0 +1,242 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_0s"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra/tier_1s"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var IPSecVpnServiceIkeLogLevelTypes = []string{
+	model.IPSecVpnService_IKE_LOG_LEVEL_DEBUG,
+	model.IPSecVpnService_IKE_LOG_LEVEL_INFO,
+	model.IPSecVpnService_IKE_LOG_LEVEL_WARN,
+	model.IPSecVpnService_IKE_LOG_LEVEL_ERROR,
+	model.IPSecVpnService_IKE_LOG_LEVEL_EMERGENCY,
+}
+
+func resourceNsxtPolicyIPSecVpnService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyIPSecVpnServiceCreate,
+		Read:   resourceNsxtPolicyIPSecVpnServiceRead,
+		Update: resourceNsxtPolicyIPSecVpnServiceUpdate,
+		Delete: resourceNsxtPolicyIPSecVpnServiceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
```

Comment:
`ImportStatePassthrough` works well when there is a single object ID that can serve as basis for Read function (behind the scenes, import calls the Read function and initializes the object). For resources in this PR, however, single ID is not good enough, since gwID and sometimes more IDs are required for the Get API call. In such cases, we use custom import functions (example https://github.com/vmware/terraform-provider-nsxt/blob/master/nsxt/resource_nsxt_policy_nat_rule.go#L45)